synology_dsm/synology_dsm.py

```diff
@@ -232,14 +233,17 @@ def _request(
 
         return response
 
-    def _execute_request(self, method, url, **kwargs):
+    def _execute_request(self, method, url, params, **kwargs):
         """Function to execute and handle a request."""
         # Execute Request
         try:
             if method == "GET":
-                resp = self._session.get(url, **kwargs)
+                encoded_params = "&".join(
+                    "%s=%s" % (key, quote(value)) for key, value in params.iteritems()
+                )  # pylint: disable=unexpected-keyword-arg
+                resp = self._session.get(url, params=encoded_params, **kwargs)

```

Comment:
```suggestion
                resp = self._session.get(url, params=urlencode(params), **kwargs)
```

Reply:
Sorry I accidentally put this back in the lint fix.

Reply:
In order to encode space into `%20`, we need to do `urllib.parse.urlencode(params, quote_via=urllib.parse.quote)` in Python 3.5+. Without the `quote_via` option, `urlencode` will use `quote_plus` which encodes space into `+`. I did some test and seems quote_via is not available in both Python 3.4 and Python 2 even import from the future package. So I guess in order to support those versions, we still need to use the quote.

Reply:
Not sure if there is a better workaround. The current version is tested and working on Python 2.7, 3.4, and 3.8.

Reply:
Indeed